sycl/CMakeLists.txt

```diff
@@ -389,13 +389,16 @@ add_custom_target(sycl-compiler
           clang-offload-extract
           clang-offload-packager
           clang-linker-wrapper
+          compiler-rt
```

Comment:
Do other projects enable profiling unconditionally in the build too?

Reply:
> I think it makes sense to do this unconditionally so that the related functionality in the SYCL runtime can be tested as part of our regular end-to-end testing process.

I'd argue that that would be a reason to enable it in `configure.py` under `--ci-defaults`, but I'd still think that this unconditional dependency here is too much.

That said, if others are ok, I'm not going to block PR on this.